src/features/events/components/LocationModal/Map.tsx

```diff
@@ -0,0 +1,215 @@
+  return children(map);
+};
+
+const Map: FC<MapProps> = ({
+  locations,
+  locationId,
+  onMapClose,
+  onSelectLocation,
+}) => {
+  const messages = useMessages(messageIds);
+  const [searchString, setSearchString] = useState('');
+  const [selectedLocationId, setSelectedLocationId] = useState(
+    locationId || undefined
+  );
+
+  const bounds = latLngBounds(
+    locations.map((location) => [location.lat, location.lng])
+  );
+
+  const fuse = new Fuse(locations, {
+    keys: ['title'],
+    threshold: 0.4,
+  });
+
+  const filteredLocations = searchString
+    ? fuse.search(searchString).map((fuseResult) => fuseResult.item)
+    : locations;
+
+  const selectedLocation = locations.find(
+    (location) => location.id === selectedLocationId
+  );
+
+  return (
+    <MapContainer
+      bounds={
+        selectedLocation
+          ? latLngBounds([[selectedLocation.lat, selectedLocation.lng]])
+          : bounds
+      }
+      style={{
+        height: '80vh',
+        width: '100%',
+      }}
+    >
+      <MapProvider>
+        {(map) => (
+          <>
+            <Box
+              sx={{
+                bottom: selectedLocation ? 1 : undefined,
+                display: 'flex',
+                justifyContent: 'flex-end',
+                justifySelf: 'flex-end',
+                margin: 2,
+                position: 'absolute',
+                right: 1,
+                top: 1,
+                width: '30%',
+                zIndex: 1000,
+              }}
+            >
+              {!selectedLocation && (
+                <Autocomplete
+                  disableClearable
+                  fullWidth
+                  onChange={(ev, value) => {
+                    const location = locations.find(
+                      (location) => location.title === value
+                    );
+                    if (!location?.lat || !location?.lng) {
+                      return;
+                    }
+                    map.setView(latLng(location.lat, location.lng), 17);
+                  }}
+                  onInputChange={(ev, value) => {
+                    setSearchString(value || '');
+                    setSelectedLocationId(
+                      locations.find((location) => location.title === value)
+                        ?.id || undefined
+                    );
+                  }}
+                  options={locations.map((location) => location.title)}
+                  renderInput={(params) => (
+                    <TextField
+                      {...params}
+                      label={messages.locationModal.searchBox()}
+                      onChange={(ev) => setSearchString(ev.target.value)}
+                      sx={{
+                        backgroundColor: 'white',
+                        borderRadius: '5px',
+                      }}
+                    />
+                  )}
+                />
+              )}
+              {selectedLocation && (
+                <Box
```

Comment:
I think it will make sense to move this into a separate component. No need to do it in this PR, but it's likely something that you'll want to do in an upcoming change, as it gets even more complex.

Reply:
Yep, these are being made separate components in my current work! :D